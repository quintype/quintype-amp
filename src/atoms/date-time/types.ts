export interface DateTimeProps {
  formattedDate: string;
  prepend?: string;
}
