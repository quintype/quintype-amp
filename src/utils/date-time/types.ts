export interface HumanizedDateTimeTypes {
  dateFormat: string;
  timeZone: string;
  timeStamp: number | Date;
}
