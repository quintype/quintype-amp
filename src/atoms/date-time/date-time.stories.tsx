import * as React from "react";
import { storiesOf } from "@storybook/react";
import { DateTime } from "./date-time";
import { config, textStory } from "../../__fixtures__";
import Layout from "../layout";

storiesOf("DateTime", module)
  .addDecorator((story) => (
    <Layout story={textStory} config={config}>
      {story()}
    </Layout>
  ))
  .add("With date", () => <DateTime dateTime={1579162728583} />)
  .add("With date and time", () => <DateTime dateTime={1579162728583} showTime={true} />)
  .add("With custom date format", () => <DateTime dateTime={1579162728583} formatString="dd MMMM yyyy" />);
