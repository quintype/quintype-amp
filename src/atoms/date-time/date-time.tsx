import React from "react";
import { DateTimeProps } from "./types";
import styled from "styled-components";

export const StyledTime = styled.time`
  color: ${(props) => props.theme.color.mono5};
  font-family: ${(props) => props.theme.font.family.secondary};
  font-size: ${(props) => props.theme.font.size.tiny};
  margin: 0;
`;

export const DateTime = ({ prepend, formattedDate }: DateTimeProps) => (
  <StyledTime dateTime={formattedDate}>
    {prepend && prepend}
    {formattedDate}
  </StyledTime>
);
