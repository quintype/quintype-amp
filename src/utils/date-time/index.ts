import { format, utcToZonedTime } from "date-fns-tz";
import { HumanizedDateTimeTypes } from "./types";

export const getHumanizedDateTime = ({ dateFormat, timeZone, timeStamp }: HumanizedDateTimeTypes) => {
  if (!dateFormat || !timeZone || !timeStamp)
    throw new Error("Required params not passed to getHumanizedDateTime function");
  const timeZonedTime = utcToZonedTime(timeStamp, timeZone);
  return format(timeZonedTime, dateFormat, { timeZone });
};
