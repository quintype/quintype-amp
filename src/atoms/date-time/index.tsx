import { DateTime } from "./date-time";
export { DateTime };
