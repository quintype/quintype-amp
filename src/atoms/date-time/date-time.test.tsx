import React from "react";
import { DateTime } from "./date-time";
import { shallow } from "enzyme";

describe("DateTime", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(<DateTime formattedDate="14 June 2017" prepend="Published: " />);
    expect(wrapper).toMatchSnapshot();
  });
  it("should render", () => {
    const wrapper = shallow(<DateTime formattedDate="14 June 2017" />);
    expect(wrapper.text()).toBe("14 June 2017");
  });
  it("should prepend text", () => {
    const wrapper = shallow(<DateTime formattedDate="14 June 2017" prepend="Published: " />);
    expect(wrapper.text()).toBe("Published: 14 June 2017");
  });
});
