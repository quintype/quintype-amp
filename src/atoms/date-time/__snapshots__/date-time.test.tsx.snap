// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateTime should match snapshot 1`] = `
<styled.time
  dateTime="14 June 2017"
>
  Published: 
  14 June 2017
</styled.time>
`;
